### PR TITLE
Makes the dropwall probably viable

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

The dropwall now lasts a minute rather than 12 seconds. Having cover for such a short time is practical, and it doesnt block movement, so its fine to last that long.

Dropwall has some armor now, so it doesnt die as fast as before. No melee armor.

The dropwall generator will survive explosions a little better, though a maxcap will still destroy it. Shields are uneffected.

The dropwall now automatically deploys when thrown when armed, rather than waiting for the timer to run out.

The dropwall now has an automatic mode, where it tries to orient the wall to protect you. Throwing it north will block north, south will block south, ect.

## Why It's Good For The Game

Yeah the dropwall is niche. Instant directional cover is good, but it lasts far too little time to hold a position, takes too long to deploy for quick combat, and falls apart rapidly. As well as being clunky to use.

This PR should address those issues. It will now last a minute, while not being forever, should last long enough for most situations.

The faster deploy / smart deploy makes it much better to use in the middle of combat. It doesnt stop a person from running up to you, but if someone is shooting at you one can get the cover up fast.

Also a bit more tankyness. This might be dropped, if it becomes too strong (doubt)

## Images of changes

https://github.com/ParadiseSS13/Paradise/assets/52090703/aec6f6de-b6bf-4c6f-b106-c808e02f1ee0

## Testing

throwing dropwalls a bunch.

## Changelog
:cl: Qwertytoforty, Contrabang
tweak: Dropwalls have bullet / laser / energy armor now.
tweak: Dropwalls last a minute now.
tweak: Dropwalls now have a smart deploy mode, and if armed will deploy as soon as they finish being thrown.
/:cl:

### CODE REVIEW

File: code/game/machinery/deployable.dm [hunk 1]
```diff
@@ -227,6 +229,7 @@
 	desc = "A temporary deployable energy shield powered by a generator. Breaking the generator will destroy all the shields connected to it."
 	icon = 'icons/obj/dropwall.dmi'
 	icon_state = "dropwall_dead" //sprite chosen in init
+	armor = list(melee = 0, bullet = 50, laser = 50, energy = 50, bomb = 10, bio = 100, rad = 100, fire = 10, acid = 0) // Copied from the security barrier, but no melee armor
```
Comment (Henri215) [inline code/game/machinery/deployable.dm:232]:
```suggestion
	armor = list(MELEE = 0, BULLET = 50, LASER = 50, ENERGY = 50, BOMB = 10, BIO = 100, RAD = 100, FIRE = 10, ACID = 0) // Copied from the security barrier, but no melee armor
```

File: code/game/machinery/deployable.dm [hunk 2]
```diff
@@ -6,7 +6,9 @@
 #define WOOD 2
 #define SAND 3
 
-#define DROPWALL_UPTIME 12
+#define DROPWALL_UPTIME 1 MINUTE
```
Comment (Contrabang) [inline code/game/machinery/deployable.dm:9]:
```suggestion
#define DROPWALL_UPTIME 1 MINUTES
```
im dumb and gave you the wrong MINUTE

File: code/game/machinery/deployable.dm [hunk 3]
```diff
@@ -271,24 +274,46 @@
 	icon = 'icons/obj/dropwall.dmi'
 	icon_state = "dropwall"
 	item_state = "grenade"
-	mode = NORTH
+	mode = AUTO
 	var/uptime = DROPWALL_UPTIME SECONDS
```
Comment (Henri215) [inline code/game/machinery/deployable.dm:278]: Could remove this var and use the define directly since we are doing "MINUTES" there.

Comment (Qwertytoforty) [reply code/game/machinery/deployable.dm:278]: Could but it depends if we want to change it again

**Comment by Carthusia** [comment]
You know, i kinda forgot dropwalls were a thing, i dont think ive ever seen them used in combat, not even against nuke ops

File: code/game/machinery/deployable.dm [hunk 4]
```diff
@@ -271,24 +274,46 @@
...
+/obj/item/grenade/barrier/dropwall/attack_self(mob/user)
+	. = ..()
+	armer = user
+
 
-	to_chat(user, "[src] is now in [dir2text(mode)] mode.")
+/obj/item/grenade/barrier/dropwall/end_throw()
+	if(active)
+		addtimer(CALLBACK(src, PROC_REF(prime)), 1) //Wait for the throw to fully end
+	return
```
Comment (Sirryan2002) [inline code/game/machinery/deployable.dm:307]:
```suggestion
```
dangling return

File: code/game/machinery/deployable.dm [hunk 5]
```diff
@@ -355,6 +380,12 @@
 	else
 		qdel(src)
 
+/obj/structure/dropwall_generator/ex_act(severity)
+	if(protected && severity > 1) //We would throw the explosion at the shield, but it is already getting hit
+		return
+	else
+		qdel(src)
```
Comment (Sirryan2002) [inline code/game/machinery/deployable.dm:387]:
```suggestion
	qdel(src)
```
you're already returning above so an else statement is not needed

File: code/game/machinery/deployable.dm [hunk 6]
```diff
@@ -271,24 +274,45 @@
 	icon = 'icons/obj/dropwall.dmi'
 	icon_state = "dropwall"
 	item_state = "grenade"
-	mode = NORTH
+	mode = AUTO
 	var/uptime = DROPWALL_UPTIME SECONDS
```
Comment (Sirryan2002) [inline code/game/machinery/deployable.dm:278]: why are you multiplying this by seconds here? The MINUTES define already does this
```dm
#define MINUTES SECONDS*60
```

Comment (Qwertytoforty) [reply code/game/machinery/deployable.dm:278]: Originally it was 60 seconds, forgot to change it over when someone requested it to be changed to a minute

**Review by Sirryan2002 — APPROVED** [review]

**Review by farie82 — APPROVED** [review]

**Pull request merged**